server/src/server.ts

```diff
@@ -251,7 +260,17 @@ async function scanDocument(document: TextDocument, connection?: IConnection) {
   }
 }
 
-
+async function handleErrors<Result>(promise: Promise<Result>):
+    Promise<Result|undefined> {
+  try {
+    return await promise;
+  } catch (err) {
+    if (connection && !(err instanceof WarningCarryingException)) {

```

Comment:
If a WarningCarryingException is caught this is swallowed without any output etc because they will be surfaced elsewhere such as document getWarnings, right?